### ISSUE TITLE
Create custom field mappings

### ISSUE THREAD

**Issue by dsilvam** [dotCMS/core#16625]
Introduce the possibility of defining the ES mapping on a per dotCMS field basis. 

Field variable should be called `esCustomMapping` and it takes a json that contains the mapping property for the given field.

After saving/editing the field mapping, a reindex will be required.   When a new reindex is fired, we will pull a list of fields that have a custom mapping specified and will PUT those to the index one by one when the new index is created and if any of the custom mappings raise an error, we will send a SystemMessage to the UI saying the field mapping failed with details but the reindex process will continue normally and the bad mapping will be ignored.

Additionally, if the field is a relationship field, we will automatically set its type to `keyword`

Mappings look like:
```json
{
"type":  "keyword",
"ignore_above": 8191
}
```

or 
```json
{
"analyzer": "my_analyzer",
"store": "false",
"type": "text"
}
```

**Update** / Expected Behavior
- [x] When the json mapping is incorrect, we need to show a toast notification saying the json mapping is invalid. It should be a warning and never prevent from saving of the field variable
- [x] In case the field variable is applied on a field that is not marked as "System Indexed", the "System Indexed" flag should be turned on at API level.

**Comment by nollymar** [comment]
**Note to Doc:** As deleting mappings from an index is not supported by Elastic, when `esCustomMapping` field variable is removed, the custom mapping will remain being used in active indices until a full reindex is run. More info: https://www.elastic.co/guide/en/elasticsearch/reference/7.2/indices-delete-mapping.html

**Comment by nollymar** [comment]
Notifications displayed in UI:

- When a field variable is created using an invalid mapping: 

<img width="1670" alt="Screen Shot 2019-12-06 at 3 08 31 PM" src="https://user-images.githubusercontent.com/8741395/70362288-6f2fa600-1852-11ea-86d1-7eeebbd7a84d.png">

- When an error arises setting custom mapping during a reindex process (the reindex process will continue and the error will be ignored): 

<img width="1651" alt="Screen Shot 2019-12-06 at 4 49 46 PM" src="https://user-images.githubusercontent.com/8741395/70362767-1b25c100-1854-11ea-8d05-9c5487969047.png">

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/17715

**Comment by wezell** [comment]
Nolly, this looks great - one thing I would add to the notification on the invalid custom mapping message - the content type.  So the message should be something like:
`Error updating mapping for field "employee.age"` , where it is explicit in `contentTypeVar.fieldVar`

**Comment by nollymar** [comment]
I have updated the error message to include the content type var as a prefix

**Comment by dsilvam** [comment]
Internal QA Results: 

1. Valid mapping. **OK**. New mapping applied after reindexing.
2. Invalid mapping (Valid JSON, invalid analyzer) **OK**. Notification shown when reindexing.
3. Invalid mapping (invalid JSON). **Not completely OK**, **the toast message goes like "Invalid format on field variable value. Value should be a JSON object.**. Starting quote should be removed.
4. When adding a valid custom mapping field var to a field with NO "System Indexed", it's correctly marking it as "System Indexed". **OK**
5. Both legacy rel and new relationship fields are mapped as keywords. **OK**

**Comment by nollymar** [comment]
The starting quote has been removed: https://github.com/dotCMS/core/pull/17730

**Comment by dsilvam** [comment]
Notification message fixed: _Error updating index mapping for field TypeWithCustomMapping.field1. Custom mapping will be ignored for index working_20191211125758._

**Comment by bryanboza** [comment]
Fixed, now this is working on release-5.2.3. New enhancements reported here #17784
